subprojects/docs/docs.gradle

```diff
@@ -113,6 +124,33 @@ outputs.docs = files(docsDir) {
     builtBy 'javadocAll', 'userguide', 'dslHtml', 'releaseNotes'
 }
 
+tasks.withType(AsciidoctorTask) {
+    backends = ['docbook']
+    separateOutputDirs = false
+    options doctype: 'book'
+    inputs.file asciidocSanitizeStylesheet withPropertyName "sanitizerStylesheet"
+
+    extensions {
+        inlinemacro (name: "api") {
+            parent, target, attributes ->
+            def (className, method) = target.split('#') as List
+            def content = "<apilink class=\"$className\""
+            if (method) {
+                content += " method=\"${method.replaceAll(/,\s*/, ', ')}\""
```

Comment:
Performance-wise, `String` concatenation is _really_ awful in Groovy (because it's not `+`, it's `plus` which can be redefined dynamically). So I would suggest to use `StringBuilder` here.